go.mod

```diff
@@ -3,47 +3,53 @@ module k8s.io/cloud-provider-aws
+	k8s.io/cloud-provider => k8s.io/kubernetes/staging/src/k8s.io/cloud-provider v0.0.0-20201120053651-7d72ddca6e9a
+	k8s.io/cluster-bootstrap => k8s.io/kubernetes/staging/src/k8s.io/cluster-bootstrap v0.0.0-20201120053651-7d72ddca6e9a
+	k8s.io/code-generator => k8s.io/kubernetes/staging/src/k8s.io/code-generator v0.0.0-20201120053651-7d72ddca6e9a
+	k8s.io/component-base => k8s.io/kubernetes/staging/src/k8s.io/component-base v0.0.0-20201120053651-7d72ddca6e9a
+	k8s.io/component-helpers => k8s.io/kubernetes/staging/src/k8s.io/component-helpers v0.0.0-20201120053651-7d72ddca6e9a
+	k8s.io/controller-manager => k8s.io/kubernetes/staging/src/k8s.io/controller-manager v0.0.0-20201120053651-7d72ddca6e9a
+	k8s.io/cri-api => k8s.io/kubernetes/staging/src/k8s.io/cri-api v0.0.0-20201120053651-7d72ddca6e9a
+	k8s.io/csi-translation-lib => k8s.io/kubernetes/staging/src/k8s.io/csi-translation-lib v0.0.0-20201120053651-7d72ddca6e9a
+	k8s.io/kube-aggregator => k8s.io/kubernetes/staging/src/k8s.io/kube-aggregator v0.0.0-20201120053651-7d72ddca6e9a
+	k8s.io/kube-controller-manager => k8s.io/kubernetes/staging/src/k8s.io/kube-controller-manager v0.0.0-20201120053651-7d72ddca6e9a
+	k8s.io/kube-proxy => k8s.io/kubernetes/staging/src/k8s.io/kube-proxy v0.0.0-20201120053651-7d72ddca6e9a
+	k8s.io/kube-scheduler => k8s.io/kubernetes/staging/src/k8s.io/kube-scheduler v0.0.0-20201120053651-7d72ddca6e9a
+	k8s.io/kubectl => k8s.io/kubernetes/staging/src/k8s.io/kubectl v0.0.0-20201120053651-7d72ddca6e9a
+	k8s.io/kubelet => k8s.io/kubernetes/staging/src/k8s.io/kubelet v0.0.0-20201120053651-7d72ddca6e9a
+	k8s.io/legacy-cloud-providers => k8s.io/kubernetes/staging/src/k8s.io/legacy-cloud-providers v0.0.0-20201120053651-7d72ddca6e9a
+	k8s.io/metrics => k8s.io/kubernetes/staging/src/k8s.io/metrics v0.0.0-20201120053651-7d72ddca6e9a
+	k8s.io/mount-utils => k8s.io/kubernetes/staging/src/k8s.io/mount-utils v0.0.0-20201120053651-7d72ddca6e9a
+	k8s.io/sample-apiserver => k8s.io/kubernetes/staging/src/k8s.io/sample-apiserver v0.0.0-20201120053651-7d72ddca6e9a
 )
 
 require (
-	github.com/aws/aws-sdk-go v1.28.2
-	github.com/golang/mock v1.3.1
-	github.com/google/go-cmp v0.4.0
-	github.com/google/uuid v1.1.1
-	github.com/spf13/cobra v1.0.0
+	github.com/aws/aws-sdk-go v1.35.24
+	github.com/golang/mock v1.4.1
+	github.com/google/go-cmp v0.5.2
+	github.com/google/uuid v1.1.2
+	github.com/spf13/cobra v1.1.1
 	github.com/spf13/pflag v1.0.5
-	github.com/stretchr/testify v1.4.0
+	github.com/stretchr/testify v1.6.1
 	gopkg.in/gcfg.v1 v1.2.0
 	k8s.io/api v0.0.0
 	k8s.io/apimachinery v0.0.0
 	k8s.io/apiserver v0.0.0
 	k8s.io/client-go v0.0.0
 	k8s.io/cloud-provider v0.0.0
 	k8s.io/component-base v0.0.0
+	k8s.io/component-helpers v0.0.0
+	k8s.io/controller-manager v0.0.0
 	k8s.io/klog v1.0.0
-	k8s.io/klog/v2 v2.2.0
-	k8s.io/kubernetes v0.0.0-20201023141757-9e8ad8ce9d8a
+	k8s.io/klog/v2 v2.4.0
+	k8s.io/kubernetes v1.20.0-beta.2.0.20201120053651-7d72ddca6e9a
```

Comment:
Are we only having this because you're pulling in k8s.io/kubernetes/staging ? Seems like we could fix that by using the public versions of the staging repo. Thats would allow you to add a rule preventing any attempt to depend on K8s.io private APIs.

Reply:
Yeah agreed, I _think_ we can remove this module by bumping all the other `k8s.io/*` modules to `v0.20.0-beta.2` (e.g. https://github.com/kubernetes/cloud-provider/releases/tag/v0.20.0-beta.2)